flake.nix

```diff
@@ -13,56 +13,76 @@
 
   inputs = {
     nixpkgs.url = "github:NixOS/nixpkgs/nixos-unstable-small";
-    flake-utils.url = "github:numtide/flake-utils";
-
-    flake-compat = {
-      url = "github:edolstra/flake-compat";
+    monad-schedule = {
+      url = "github:turion/monad-schedule";
       flake = false;
     };
-
-    haskell-flake-utils = {
-      url = "github:ivanovs-4/haskell-flake-utils";
-      inputs.flake-utils.follows = "flake-utils";
+    time-domain = {
+      url = "github:turion/time-domain";
+      flake = false;
     };
   };
 
-outputs = { self, nixpkgs, flake-utils, haskell-flake-utils, flake-compat, ... }:
-  haskell-flake-utils.lib.simpleCabalProject2flake {
-    inherit self nixpkgs;
+  outputs = inputs:
+    with builtins;
+    let
+      lib = inputs.nixpkgs.lib;
+      pnames = [
+        "automaton"
+        "rhine"
+        "rhine-bayes"
+        "rhine-examples"
+        "rhine-gloss"
+        "rhine-terminal"
+      ];
+      supportedGhcs = [ "ghc92" "ghc94" "ghc96" "ghc98" ];
+      hpsFor = pkgs:
+        lib.filterAttrs (name: _: elem name supportedGhcs) pkgs.haskell.packages
+        // { default = pkgs.haskellPackages; };
+      rhinePackages = hfinal: hprev:
+        lib.genAttrs pnames (pname: hfinal.callCabal2nix pname ./${pname} { });
+      overlay = final: prev: {
+        haskell = prev.haskell // {
+          packageOverrides = with prev.haskell.lib.compose; lib.composeManyExtensions [
+            prev.haskell.packageOverrides
+            rhinePackages
+            (hfinal: hprev: {
+              monad-bayes = markUnbroken hprev.monad-bayes;
+              monad-schedule = hfinal.callCabal2nix "monad-schedule" inputs.monad-schedule { };
+              time-domain = hfinal.callCabal2nix "time-domain" inputs.time-domain { };
+            })
+            (hfinal: hprev: lib.optionalAttrs (lib.versionOlder hprev.ghc.version "9.4") {
+              time-domain = doJailbreak hprev.time-domain;
+            })
```

Comment:
I'd rather use callHackageDirect here than refer to github. After all, I'm maintaining a build with Debian + cabal in another CI job as well, so overrides will only be necessary whenever a Hackage version is not on stackage/nixpkgs.

If `callHackageDirect` is a poor choice for some other reason, we could still use Github and specific tags corresponding to versions. But I don't want to use `master` for these dependencies because I might push a breaking version.

Reply:
I went with `callHackageDirect`, since that's what we had before.

My original thinking was that we might not want to keep updating the flake with new hashes every time these dependencies update. Ideally that's something that flake inputs can automate.